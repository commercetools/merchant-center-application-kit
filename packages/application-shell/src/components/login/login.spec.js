import React from 'react';
// import { LOGOUT_REASONS } from '@commercetools-frontend/constants';
import {
  renderAppWithRedux,
  wait,
  waitForElement,
  fireEvent,
} from '../../test-utils';
import Login from './login';

jest.mock('@commercetools-frontend/storage');

const createTestProps = props => ({
  location: { query: {} },
  history: {
    push: jest.fn(),
  },
  ...props,
});

// TODO: write new tests for the login interaction

describe('reset password flow', () => {
  beforeEach(() => {
    window.location.replace = jest.fn();
  });
  it('should open a dialog and redirect to the AC', async () => {
    const props = createTestProps();
    const { getByText } = renderAppWithRedux(<Login {...props} />, {
      environment: { adminCenterUrl: 'http://ac.com' },
    });

    await waitForElement(() => getByText(/Sign in to your account/));

    fireEvent.click(getByText(/Forgot password/));

    await waitForElement(() =>
      getByText(
        /We are redirecting you to the Forgot Password form in 3 seconds/
      )
    );

    await wait(() => {
      expect(window.location.replace).toHaveBeenCalledWith(
        'http://ac.com/reset-password'
      );
    });
  });
});
