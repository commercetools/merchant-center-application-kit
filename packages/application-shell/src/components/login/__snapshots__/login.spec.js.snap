// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`rendering should match tree 1`] = `
<PublicPageContainer>
  <LoginBox>
    <form
      className="form"
      method="post"
      onChange={[Function]}
      onSubmit={[Function]}
    >
      <fieldset
        className="fieldset"
      >
        <Stack
          alignItems="stretch"
          scale="m"
        >
          <legend
            className="title"
          >
            <Title>
              <FormattedMessage
                defaultMessage="Sign in to your account"
                description="The title message for the login form"
                id="LoginForm.title"
                values={Object {}}
              />
            </Title>
          </legend>
          <Stack
            alignItems="stretch"
            scale="m"
          >
            <Stack
              alignItems="stretch"
              scale="s"
            >
              <label
                className="label"
                htmlFor="email"
              >
                <FormattedMessage
                  defaultMessage="Email"
                  description="The label for the login input field (email)"
                  id="LoginForm.email"
                  values={Object {}}
                />
              </label>
              <input
                className="input-text"
                name="email"
                onKeyDown={[Function]}
                type="text"
              />
            </Stack>
            <Stack
              alignItems="stretch"
              scale="s"
            >
              <label
                className="label"
                htmlFor="password"
              >
                <FormattedMessage
                  defaultMessage="Password"
                  description="The label for the login input field (password)"
                  id="LoginForm.password"
                  values={Object {}}
                />
              </label>
              <input
                className="input-text"
                name="password"
                onKeyDown={[Function]}
                type="password"
              />
            </Stack>
          </Stack>
          <div>
            <PrimaryButton
              isToggleButton={false}
              label="LoginForm.signin"
              onClick={[Function]}
              size="big"
              tone="primary"
            />
          </div>
          <Contdown
            isDisabled={false}
            redirectTo={[Function]}
          >
            <Component />
          </Contdown>
        </Stack>
      </fieldset>
    </form>
  </LoginBox>
</PublicPageContainer>
`;
